### PR TITLE
Validation for New Access Management - Phase 2 (AST-64636)

### PR DESCRIPTION
**Pull request by tamarleviCm**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

> Describe the purpose of this PR along with any background information and the impacts of the proposed change.

### References

> Include supporting link to GitHub Issue/PR number

### Testing

> Describe how this change was tested. Be specific about anything not tested and reasons why. If this solution has unit and/or integration testing, tests should be added for new functionality and existing tests should complete without errors.
>
> Please include any manual steps for testing end-to-end or functionality not covered by unit/integration tests.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] I have updated the CLI help for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: test/integration/scan_test.go [hunk 1]
```diff
@@ -870,9 +872,9 @@ func executeScanAssertions(t *testing.T, projectID, scanID string, tags map[stri
 
 func createScan(t *testing.T, source string, tags map[string]string) (string, string) {
 	if isFFEnabled(t, wrappers.ContainerEngineCLIEnabled) {
-		return executeCreateScan(t, getCreateArgs(source, tags, "sast , sca , iac-security , api-security, scs,container-security"))
+		return executeCreateScan(t, getCreateArgs(source, tags, "sast , sca , iac-security , api-security, container-security"))
```
Comment (tamarleviCm) [inline test/integration/scan_test.go:875]: remove

File: test/integration/predicate_test.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ func TestSastUpdateAndGetPredicatesForSimilarityId(t *testing.T) {
 		flag(params.ScanIDFlag), scanID, flag(params.TargetFormatFlag), printer.FormatJSON,
 		flag(params.TargetPathFlag), resultsDirectory,
 		flag(params.TargetFlag), fileName,
+		flag(params.DebugFlag),
```
Comment (tamarleviCm) [inline test/integration/predicate_test.go:32]: remove

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -46,14 +46,14 @@ jobs:
...
         env:
-          CX_BASE_URI: ${{ secrets.CX_BASE_URI }}
-          CX_CLIENT_ID: ${{ secrets.CX_CLIENT_ID }}
-          CX_CLIENT_SECRET: ${{ secrets.CX_CLIENT_SECRET }}
-          CX_BASE_AUTH_URI: ${{ secrets.CX_BASE_AUTH_URI }}
-          CX_AST_USERNAME: ${{ secrets.CX_AST_USERNAME }}
-          CX_AST_PASSWORD: ${{ secrets.CX_AST_PASSWORD }}
-          CX_APIKEY: ${{ secrets.CX_APIKEY }}
-          CX_TENANT: ${{ secrets.CX_TENANT }}
+          CX_BASE_URI: ${{ secrets.CX_BASE_URI_ACCESS }}
```
Comment (tamarleviCm) [inline .github/workflows/ci.yml:49]: remove

File: test/integration/scan_test.go [hunk 4]
```diff
@@ -296,15 +296,17 @@ func TestScanCreateEmptyProjectName(t *testing.T) {
...
 		flag(params.ApplicationName), "my-application",
-		flag(params.ProjectName), getProjectNameForScanTests(),
+		flag(params.ProjectName), projectName,
 		flag(params.SourcesFlag), ".",
 		flag(params.ScanTypes), params.IacType,
 		flag(params.BranchFlag), "dummy_branch",
+		flag(params.DebugFlag),
 	}
-
+	defer deleteProject(t, projectId)
```
Comment (OrShamirCM) [inline test/integration/scan_test.go:309]: @BenAlvo1  we deleted the defer delete for everything, right?
@miryamfoiferCX , ן think we shou;d remove it

File: test/integration/scan_test.go [hunk 5]
```diff
@@ -1871,14 +1874,16 @@ func addSCSDefaultFlagsToArgs(args *[]string) {
 func TestCreateScanAndValidateCheckmarxDomains(t *testing.T) {
 	wrappers.Domains = make(map[string]struct{})
 	_, _ = executeCreateScan(t, getCreateArgsWithGroups(Zip, Tags, Groups, "iac-security"))
-	usedDomainsInTests := []string{"deu.iam.checkmarx.net", "deu.ast.checkmarx.net"}
-	validateCheckmarxDomains(t, usedDomainsInTests)
+	baseUrl, _ := wrappers.GetURL("", "")
+	authUri, _ := wrappers.GetAuthURI()
+	usedDomainsFromConfig := []string{baseUrl, authUri}
+	validateCheckmarxDomains(t, usedDomainsFromConfig)
```
Comment (OrShamirCM) [inline test/integration/scan_test.go:1880]: Nice!

File: test/integration/scan_test.go [hunk 6]
```diff
@@ -1871,14 +1874,16 @@ func addSCSDefaultFlagsToArgs(args *[]string) {
...
 }
 
 func validateCheckmarxDomains(t *testing.T, usedDomainsInTests []string) {
 	usedDomains := wrappers.Domains
 	for domain, _ := range usedDomains {
-		assert.Assert(t, slices.Contains(usedDomainsInTests, domain), "Domain "+domain+" not found in used domains")
+		assert.Assert(t, utils.Contains(usedDomainsInTests, domain), "Domain "+domain+" not found in used domains")
```
Comment (OrShamirCM) [inline test/integration/scan_test.go:1886]: what is the different, why do we have continas in util?

File: test/integration/util.go [hunk 7]
```diff
@@ -55,6 +55,7 @@ func flag(f string) string {
 }
 
 func getProjectNameForTest() string {
+	projectNameRandom = uuid.New().String()
```
Comment (OrShamirCM) [inline test/integration/util.go:58]: how did it work before?

File: test/integration/scan_test.go (hunk 4 above)
Comment (miryamfoiferCX) [reply test/integration/scan_test.go:309]: but here we create a new project, not using the same as other test, so when will we delete it?

File: test/integration/scan_test.go (hunk 6 above)
Comment (miryamfoiferCX) [reply test/integration/scan_test.go:1886]: checking also sun strings

File: test/integration/project_test.go [hunk 8]
```diff
@@ -40,7 +40,7 @@ func TestProjectsE2E(t *testing.T) {
 	assert.Equal(t, len(response), 1, "Total projects should be 1")
 	assert.Equal(t, response[0].ID, projectID, "Project ID should match the created project")
 
-	project := showProject(t, projectID)
+	project := response[0]
```
Comment (AlvoBen) [inline test/integration/project_test.go:43]: show project is another functionality we are testing isn't it? why removing it

File: test/integration/util.go (hunk 7 above)
Comment (AlvoBen) [reply test/integration/util.go:58]: This is effect the deletion of tests in the end. revert this change

File: internal/services/projects.go [hunk 9]
```diff
@@ -34,7 +34,7 @@ func FindProject(
 	params := make(map[string]string)
 	params["names"] = projectName
 	resp, _, err := projectsWrapper.Get(params)
-	if err != nil {
+	if err != nil || resp == nil {
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply internal/services/projects.go:37]: @tamarleviCm

File: test/integration/project_test.go (hunk 8 above)
Comment (miryamfoiferCX) [reply test/integration/project_test.go:43]: @tamarleviCm

File: test/integration/scan_test.go (hunk 4 above)
Comment (miryamfoiferCX) [reply test/integration/scan_test.go:309]: @tamarleviCm

File: test/integration/project_test.go (hunk 8 above)
Comment (tamarleviCm) [reply test/integration/project_test.go:43]: This test is not for testing this command, use it to get the data again (seems unnecessary anyway)
When the flag is on, the group data is not returned because the behavior is different in Access Management
Therefore we used what is returned from the listProjectByID

File: internal/services/projects.go (hunk 9 above)
[the remark this answers is not in the document]
Comment (OrShamirCM) [reply internal/services/projects.go:37]: maybe to create new err?

[the remark this answers is not in the document]
Comment (OrShamirCM) [reply internal/services/projects.go:37]: or if the project is nil so creating new? so don't need to create err

File: internal/wrappers/export-http.go [hunk 10]
```diff
@@ -166,18 +166,26 @@ func (e *ExportHTTPWrapper) GetScaPackageCollectionExport(fileURL string) (*ScaP
...
 		return nil, err
 	}
+
 	resp, err := SendHTTPRequestByFullURL(http.MethodGet, fileURL, http.NoBody, true, viper.GetUint(commonParams.ClientTimeoutKey), accessToken, true)
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 
+	body, err := io.ReadAll(resp.Body)
```
Comment (OrShamirCM) [inline internal/wrappers/export-http.go:176]: merge with ben changes, remove from your code

File: test/integration/project_test.go (hunk 8 above)
Comment (OrShamirCM) [reply test/integration/project_test.go:43]: Need to revert this change. need to fix showproject command to have groups with new access management

File: test/integration/scan_test.go [hunk 11]
```diff
@@ -295,19 +295,23 @@ func TestScanCreateEmptyProjectName(t *testing.T) {
 	assertError(t, err, "Project name is required") // Creating a scan with empty project name should fail
 }
 
+/*
```
Comment (OrShamirCM) [inline test/integration/scan_test.go:298]: let's wait for the bug. need to remove comment

File: test/integration/project_test.go [hunk 12]
```diff
@@ -157,6 +157,10 @@ func TestProjectBranches(t *testing.T) {
 
 func createProject(t *testing.T, tags map[string]string, groups []string) (string, string) {
 	projectName := getProjectNameForTest() + "_for_project"
```
Comment (OrShamirCM) [inline test/integration/project_test.go:159]: create new project - need to assign with the project name - GenerateRandomProjectNameForScan()

File: test/integration/project_test.go [hunk 13]
```diff
@@ -157,6 +157,10 @@ func TestProjectBranches(t *testing.T) {
 
 func createProject(t *testing.T, tags map[string]string, groups []string) (string, string) {
```
Comment (OrShamirCM) [inline test/integration/project_test.go:158]: please delete getProjectNameForTest

Comment (miryamfoiferCX) [reply test/integration/project_test.go:158]: I made a refactor in **_getProjectNameForTest_**, now it's using **_GenerateRandomProjectNameForScan_**
so now we generate a project name only one time per run, and get it for each test that needs a project name
only test that needs a new project name will call to **_GenerateRandomProjectNameForScan_** and this will write it also to the file as we are doing for the first project that have been created for all tests....

File: test/integration/project_test.go (hunk 12 above)
Comment (miryamfoiferCX) [reply test/integration/project_test.go:159]: same response as above - refactor was made in **_getProjectNameForTest_** so now it's working as expected for all tests

File: internal/wrappers/export-http.go (hunk 10 above)
Comment (miryamfoiferCX) [reply internal/wrappers/export-http.go:176]: @OrShamirCM 
can you please check what is the status of the fix by SCA?

Comment (miryamfoiferCX) [reply internal/wrappers/export-http.go:176]: @OrShamirCM what is the status of the fix by SCA?

Comment (OrShamirCM) [reply internal/wrappers/export-http.go:176]: SCA done. Please remove the BOM code

File: test/integration/project_test.go (hunk 12 above)
Comment (OrShamirCM) [reply test/integration/project_test.go:158]: so now you creating a new project but with an exciting name? it is supposed to fail or only update the project. I think you should use here GenerateRandomProjectNameForScan

File: internal/services/projects.go [hunk 14]
```diff
@@ -70,6 +68,22 @@ func FindProject(
 	return projectID, nil
 }
 
+func GetProjectsCollectionByProjectName(projectName string, projectsWrapper wrappers.ProjectsWrapper) (*wrappers.ProjectsCollectionResponseModel, error) {
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply internal/services/projects.go:71]: ProjectsCollection is the name of the returns module

File: test/integration/project_test.go (hunk 12 above)
Comment (miryamfoiferCX) [reply test/integration/project_test.go:158]: we shouldn't random a new name here, bacause we don't want to create a newproject
only if we want a new project we use it

File: internal/services/projects.go (hunk 14 above)
[the remark this answers is not in the document]
Comment (OrShamirCM) [reply internal/services/projects.go:71]: why by name and not ID?

[the remark this answers is not in the document]
Comment (OrShamirCM) [reply internal/services/projects.go:71]: we need to make sure we get only 1 project even if it'll find 2 names start with the same prefix

File: test/integration/import_test.go [hunk 15]
```diff
@@ -12,13 +12,9 @@ import (
 )
 
 func TestImport_ImportSarifFileWithCorrectFlags_CreateImportSuccessfully(t *testing.T) {
-
-	projectId, projectName := createProject(t, nil, nil)
-	defer deleteProject(t, projectId)
-
 	args := []string{
 		"utils", "import",
-		flag(params.ProjectName), projectName,
```
Comment (miryamfoiferCX) [inline test/integration/import_test.go:21]: revert this changes

File: test/integration/util.go [hunk 16]
```diff
@@ -12,7 +13,11 @@ import (
 	"gotest.tools/assert"
 )
 
-var projectNameRandom = uuid.New().String()
```
Comment (miryamfoiferCX) [inline test/integration/util.go:15]: revert this test

File: test/integration/project_test.go (hunk 12 above)
Comment (OrShamirCM) [reply test/integration/project_test.go:158]: @AlvoBen - please do the CR for all the tests

File: internal/services/projects_test.go [hunk 17]
```diff
@@ -279,3 +280,67 @@ func Test_updateProject(t *testing.T) {
...
+		projectsWrapper wrappers.ProjectsWrapper
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    *wrappers.ProjectsCollectionResponseModel
+		wantErr bool
+	}{
+		{
+			name: "Group exists",
```
Comment (OrShamirCM) [inline internal/services/projects_test.go:296]: write name in when_should convention (bdd)
whenCallingX_withY_ShouldbeZ

Comment (OrShamirCM) [reply internal/services/projects_test.go:296]: change from group to project

File: internal/services/projects_test.go [hunk 18]
```diff
@@ -279,3 +280,67 @@ func Test_updateProject(t *testing.T) {
...
+			args: args{
+				projectName:     "existing-group",
+				projectsWrapper: &mock.ProjectsMockWrapper{},
+			},
+			want: &wrappers.ProjectsCollectionResponseModel{
+				Projects: []wrappers.ProjectResponseModel{
+					{ID: "existing-group-id", Name: "existing-group"},
```
Comment (OrShamirCM) [inline internal/services/projects_test.go:303]: group to project

Comment (OrShamirCM) [inline internal/services/projects_test.go:303]: give many projects

File: internal/services/projects_test.go [hunk 19]
```diff
@@ -279,3 +280,67 @@ func Test_updateProject(t *testing.T) {
...
+			wantErr: false,
+		},
+		{
+			name: "Error from wrapper",
+			args: args{
+				projectName:     "error-project",
+				projectsWrapper: &mock.ProjectsMockWrapper{},
+			},
+			want:    nil,
+			wantErr: true,
```
Comment (OrShamirCM) [inline internal/services/projects_test.go:330]: we dont want error
we want response of empty project

File: internal/commands/project.go [hunk 20]
```diff
@@ -442,6 +448,21 @@ func runGetProjectByIDCommand(projectsWrapper wrappers.ProjectsWrapper) func(cmd
...
+		return wrappers.ProjectResponseModel{}, fmt.Errorf("failed to get project by name: %s", projectName)
+	}
+
+	for i := range resp.Projects {
+		project := &resp.Projects[i]
+		if project.Name == projectName {
+			return *project, nil
+		}
+	}
+	return wrappers.ProjectResponseModel{}, fmt.Errorf("project not found: %s", projectName)
```
Comment (OrShamirCM) [inline internal/commands/project.go:463]: if resp == null, return empty object and not error

**Review by OrShamirCM — COMMENTED** [review]
integration.sh is the file for the covrage, you can update the new coverage there.
Great work!

File: internal/wrappers/mock/projects-mock.go [hunk 21]
```diff
@@ -59,6 +59,20 @@ func (p *ProjectsMockWrapper) Get(params map[string]string) (
...
+	case "non-existing-project":
+		model = nil
+	case "error-project":
+		return nil, nil, fmt.Errorf("some error")
+	case "test_project1":
+		model = getProjectResponseModel("1", "test_project1", filteredTotalCount)
+	case "test_project2":
+		model = getProjectResponseModel("2", "test_project2", filteredTotalCount)
+	case "exact_project":
+		model = getProjectResponseModel("3", "exact_project", filteredTotalCount)
```
Comment (OrShamirCM) [inline internal/wrappers/mock/projects-mock.go:73]: the exact project doesn't initialise the mock with 3 different projects
you can add in getProjectResponseModel
Projects: for in count{ id: i, name: "{name}+{i}"}
then it will give you the test we wanted to test

File: internal/commands/project.go [hunk 22]
```diff
@@ -433,6 +433,9 @@ func runGetProjectByIDCommand(projectsWrapper wrappers.ProjectsWrapper) func(cmd
 		if errorModel != nil {
 			return errors.Errorf("%s: CODE: %d, %s", services.FailedGettingProj, errorModel.Code, errorModel.Message)
 		} else if projectResponseModel != nil {
+			resp := GetProjectByName(projectResponseModel.Name, projectsWrapper)
```
[the remark this answers is not in the document]
Comment (miryamfoiferCX) [reply internal/commands/project.go:436]: it's related to AM. when we access by Id we're not getting the groups
when we're getting by name we get groups

**Review by OrShamirCM — APPROVED** [review]
